### PR TITLE
feat: likely_binary_ignore

### PR DESCRIPTION
**Pull request by joshuarli**
Discovered as part of https://github.com/getsentry/pypi/pull/75. Thought it would be better to put more specific cases in `packages.ini`.

edit: not needed anymore since https://github.com/getsentry/pypi/pull/79, but gonna keep this around

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -525,6 +528,9 @@ def _likely_binary(sdist: str) -> str | None:
         if "/test/" in name or "/tests/" in name:
             continue
 
+        if any(substr in name for substr in package.likely_binary_ignore):
```
Comment (asottile-sentry) [inline build.py:531]: I think these would be better as exact matches?

File: build.py [hunk 2]
```diff
@@ -497,7 +500,7 @@ def _prefix_path(*parts: str) -> str:
             env.update(before)
 
 
-def _likely_binary(sdist: str) -> str | None:
+def _likely_binary(package: Package, sdist: str) -> str | None:
```
Comment (asottile-sentry) [inline build.py:503]: it'll be much easier to test if this takes `likely_binary_ignore` directly rather than the whole `Package`

**Review by asottile-sentry — COMMENTED** [review]
looks good -- just a few tiny things

File: build.py (hunk 1 above)
Comment (joshuarli) [reply build.py:531]: ok!

**Pull request closed without merging**